node/core/parachains-inherent/src/metrics.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use polkadot_node_subsystem_util::metrics::{self, prometheus};
+use std::convert::TryInto;
+
+#[derive(Clone)]
+struct MetricsInner {
+	/// Number of votes.
+	votes: prometheus::Counter<prometheus::U64>,
+	/// Number of disputes.
+	disputes: prometheus::Counter<prometheus::U64>,
+}
+
+/// Inherent data processing metrics for `create_inherent()`.
+#[derive(Default, Clone)]
+pub struct Metrics(Option<MetricsInner>);
+
+impl Metrics {
+	pub fn inc_votes_by(&self, votes: usize) {
+		if let Some(metrics) = &self.0 {
+			metrics.votes.inc_by(votes.try_into().unwrap_or(0));
+		}
+	}
+	pub fn inc_disputes_by(&self, disputes: usize) {
```

Comment:
I would prefer to signal the reader that this is the number of disputes sets of the _inherent_. We should also include a count for the number of actual votes included.

Reply:
Could we add a `_inherent_` to the name to make this more explicit?